### PR TITLE
[CORE] [DEVEL] Fix warnings on deprecated copy-constructors

### PR DESCRIPTION
**Pull request by smuzaffar**
Fixes deprecated warnings for DEVEL IBs.
Updated `test_precomputed_value_sort` to actually compare the results and fail in case of unexpected results.

### CODE REVIEW

File: Utilities/General/test/test_precomputed_value_sort.cpp [hunk 1]
```diff
@@ -37,8 +36,6 @@ ostream& operator<<(ostream& o, const Point* p) { return o << *p; }
 
 Point::Point(float x, float y) : X(x), Y(y) { cout << "New Point" << *this << endl; }
 
-Point::Point(const Point& p) : X(p.X), Y(p.Y) { cout << "New Point (copy)" << *this << endl; }
```
Comment (Dr15Jones) [inline Utilities/General/test/test_precomputed_value_sort.cpp:40]: Is this side-effect used in a test?

**Comment by Dr15Jones** [comment]
Since this is not actually a test (if there is a problem the program still returns 0) I thinks we should either
1. remove the test all together or
2. make the test actually check the results of the sorting.

File: Utilities/General/test/test_precomputed_value_sort.cpp (hunk 1 above)
Comment (smuzaffar) [reply Utilities/General/test/test_precomputed_value_sort.cpp:40]: does not look like

**Comment by smuzaffar** [comment]
> remove the test all together or

I am in favor of dropping it, any objections @makortel ?

**Comment by makortel** [comment]
I agree dropping the test would not make the situation worse, but in principle it would be nice to have a test for it. So I'm a little bit, but not too much, against dropping the test.

**Comment by smuzaffar** [comment]
ok, I have updated the unit tests to compare the results

**Comment by smuzaffar** [comment]
please test

**Comment by makortel** [comment]
+core

**Comment by rappoccio** [comment]
+1

**Pull request merged**